cbhwlib/cbhwlib.cpp

```diff
@@ -1262,9 +1262,11 @@ cbRESULT cbGetAoutWaveform(uint32_t channel, uint8_t  trigNum, uint16_t  * mode,
     if (!(cb_cfg_buffer_ptr[nIdx]->chaninfo[channel - 1].chancaps & cbCHAN_AOUT)) return cbRESULT_INVALIDFUNCTION;
     if (!(cb_cfg_buffer_ptr[nIdx]->chaninfo[channel - 1].aoutcaps & cbAOUT_WAVEFORM)) return cbRESULT_INVALIDFUNCTION;
     if (trigNum >= cbMAX_AOUT_TRIGGER) return cbRESULT_INVALIDFUNCTION;
-    if (channel <= cbNUM_ANALOG_CHANS) return cbRESULT_INVALIDCHANNEL;
-    channel -= (cbNUM_ANALOG_CHANS + 1); // make it 0-based
-    if (channel >= AOUT_NUM_GAIN_CHANS) return cbRESULT_INVALIDCHANNEL;
+	// TODO: Maybe we need a m_ChanIdxInType array.
+	if (cb_cfg_buffer_ptr[nIdx]->chaninfo[cbNUM_ANALOG_CHANS - 1].chancaps & cbCHANTYPE_ANAIN)
+		channel -= (cbNUM_ANALOG_CHANS + 1); // make it 0-based
+	else
+		channel -= (128 + 16 + 1);

```

Comment:
Basically what this is saying is: "if channel number 256 + 16 is analog then this must be a 2-NSP system so we'll subtract (256 + 16 + 1). Otherwise, this must be a 1-NSP system so we'll subtract (128 + 16 + 1).

There is no constant related to channel numbers defined to 128.
There is a constant for 16 (`cbNUM_ANAIN_CHANS`) so I'll switch that one at least.

Really the better solution is to use a m_ChanIdxInBuffer array (and associated getter methods). I'd really like to finish up this PR before doing that.